src/Automation/SnapshotCommand.cs

```diff
@@ -2,35 +2,57 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using Axe.Windows.Actions.Contexts;
+using Axe.Windows.Automation.Data;
 using Axe.Windows.Automation.Resources;
 using Axe.Windows.Core.Bases;
 using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Axe.Windows.Automation
 {
     /// <summary>
-    /// Class to take a snapshot (via SnapshotCommand.Execute).
+    /// Class to take a snapshot (via SnapshotCommand.Execute*).
     /// </summary>
     static class SnapshotCommand
     {
         /// <summary>
-        /// Execute the scan
+        /// Execute the scan synchronously
         /// </summary>
         /// <param name="config">A set of configuration options</param>
         /// <param name="scanTools">A set of tools for writing output files,
         /// creating the expected results format, and finding the target element to scan</param>
-        /// <returns>A SnapshotCommandResult that describes the result of the command</returns>
-        public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools scanTools)
+        /// <returns>A set of ScanResults objects that describes the result of the command</returns>
+        public static IReadOnlyCollection<ScanResults> ExecuteScanAll(Config config, IScanTools scanTools)
+        {
+            ValidateScanParameters(config, scanTools);
+
+            return GetScanAllResults(config, scanTools, CancellationToken.None);
+        }
+
+        public static Task<AsyncScan> ExecuteScanAsync(Config config, IScanTools scanTools, CancellationToken cancellationToken)
+        {
+            ValidateScanParameters(config, scanTools);
+
+            return Task.Run<AsyncScan>(() => new AsyncScan(GetScanAllResults(config, scanTools, cancellationToken).FirstOrDefault()));
```

Comment:
Creating the `AsyncScan` here works fine as long as `AsyncScan`'s constructor only takes 1 input parameter. We'll have to change this if/when we add additional input parameters, which seems likely to occur at some point. Would it make sense for `GetScanAllResults` to return a fully-formed `AsyncScan` object? The sync wrappers would then just need to return the pieces that match the existing API